stl/inc/ranges

```diff
@@ -33,10 +33,195 @@ namespace ranges {
     concept viewable_range = range<_Rng>
         && (borrowed_range<_Rng> || view<remove_cvref_t<_Rng>>);
 
+    template <class _Rng>
+    concept _Simple_view = view<_Rng> && range<const _Rng>
+        && same_as<iterator_t<_Rng>, iterator_t<const _Rng>>
+        && same_as<sentinel_t<_Rng>, sentinel_t<const _Rng>>;
+
     template <class _Ty>
     concept _Copy_constructible_object = copy_constructible<_Ty> && is_object_v<_Ty>;
     // clang-format on
 
+    namespace _Pipe {
+        template <class>
+        struct _Base;
+
+        template <class, class>
+        struct _Pipeline;
+
+        // clang-format off
+        template <class _Left, class _Right>
+        concept _Can_pipe = requires(_Left&& __l, _Right&& __r) {
+            static_cast<_Right&&>(__r)(static_cast<_Left&&>(__l));
+        };
+
+        template <class _Left, class _Right>
+        concept _Can_compose = constructible_from<remove_cvref_t<_Left>, _Left>
+            && constructible_from<remove_cvref_t<_Right>, _Right>;
+        // clang-format on
+
+        template <class _Derived>
+        struct _Base {
+            // clang-format off
+            template <class _Other>
+                requires _Can_compose<_Derived&&, _Other&&>

```

Comment:
Do you need to say `&&` here? In the `constructible_from` invocations, the first argument is `remove_cvref_t`'ed, and the second argument is `declval`'ed. Ditto below where `&&` occurs (`const&` is of course meaningful).

Reply:
No, we don't - stripped.